### ISSUE TITLE
Using dates as values

### ISSUE THREAD

**Issue by emiguevara** [darul75/ng-slider#95]
I want to use ng-slider as a date selector. Passing Unix timestamps to the slider works just perfect, and the step option is very useful too. However, I would like to transform the timestamps into date objects in order to show human readable dates on the interface.

Any experience or suggestions on how to do this? Or is it a case for a fork and pull request? Thanks!

**Comment by darul75** [comment]
Momentjs do that very well. But not sure then values displayed can be different from original value, date object in your case. Can you fork jsfiddle with your case.

**Comment by emiguevara** [comment]
I made a jsfiddle with a couple of examples https://jsfiddle.net/emiguevara/c5zf73gd/1/

If ng-model receives a date object, the slider breaks. I suppose a trick could be to make the slider work with the Unix timestamps and creating the date object just for the HTML... I hope I made myself clear, thanks!

**Comment by darul75** [comment]
Today only number or string (mandatory for range pattern) are handled for model value.
To pass object or and array of object looks fine to handle a Date model. But then display would be a simple toString method for js dates. I have to think about it but quite busy now.
You can start do play with src/ng-slider.js with a single Date object and change test made with string or number.

**Comment by emiguevara** [comment]
Thanks, looking at the code to see if I understand something... man, this is dense...

**Comment by darul75** [comment]
Yeah fu... dense because original was ;) and I have put time understand split try break again and...but that's js world crazy one. When u fork u can use index.html file in src folder with some breakpoint you may have fun or not.

**Comment by darul75** [comment]
some contributors have added 'modelLabels' option, it accept an object today but I will see to enhance it to support a function which will convert a date value to string of your choice by instance (with momentjs or whatever ;))

**Comment by kavka3** [comment]
++
